test/templates/config.ru

```diff
@@ -0,0 +1,70 @@
+$:.push File.dirname(__FILE__)
+require "fake_middleware"
+
+module Helper
+  def self.test_root
+    File.dirname(__FILE__)
+  end
+  def self.relative_path(path)
+    test_root + "/" + path
+  end
+  def self.root_path(path)
+    relative_path("../../") + path
+  end
+end
+
+# npm modules (e.g. "/node_modules/mocha/mocha.js")
+map "/node_modules" do
+  node_modules = Helper.relative_path("node_modules")
+  unless Dir.exists?(node_modules)
+    abort "missing the node_modules directory"
+  end
+  run Rack::Directory.new(node_modules)
+end
+
+# test runners
+map "/html" do
+  class TestRunners < FakeMiddleware
```

Comment:
> I've been thinking, is the inheritance issued here the best tool we have? I think we can avoid it passing req_path_regex, headers, view_paths as initializer arguments. Something in the lines of:

Yeah, that's right. How about this commit?

https://github.com/sh19910711/web-console/commit/492a4b6823617024089560fa15dc82ef83bf9666